src/arrayMethodReduce.js

```diff
@@ -5,7 +5,32 @@
  */
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+    let sum;
+    let startIndex;
+
+    // If initialvalue is NOT Provided
+    if (arguments.length === 1) {
+      sum = typeof this[0] === 'string' ? '' : 0;
+      startIndex = 0;
+    }
+
+    // If initialValue is explicitly provided as UNDEFINED
+    if (arguments.length > 1 && initialValue === undefined) {
+      sum = 'undefined';
+      startIndex = 0;
+    }
+
+    // If initialValue is Provided
+    if (arguments.length > 1 && initialValue !== undefined) {
+      sum = initialValue;
+      startIndex = 0;
+    }
```

Comment:
This can be simplified using just one if.

Reply:
required changes done